packages/hardhat/contracts/strategy/CrabMigration.sol

```diff
@@ -146,17 +150,33 @@ import {StrategyMath} from "./base/StrategyMath.sol";
          crabV2.transfer(msg.sender, amountV2ToTransfer);
      }
 
-     /** 
+      /** 
       * @notice allows users to claim crabV2 shares and flash withdraw from crabV2
+      * 
+      * @param _amountToWithdraw Amount of shares to withdraw
+      * @param _maxEthToPay maximum ETH to pay to buy back the owed wSqueeth debt
       */
-     function claimAndWithdraw() external afterMigration { 
+     function claimAndWithdraw(uint256 _amountToWithdraw, uint256 _maxEthToPay) external afterMigration { 
+        uint256 amountV1Deposited = sharesDeposited[msg.sender];
+        require(_amountToWithdraw <= amountV1Deposited, "M6");
```

Comment:
I think in sol 0.8 this should already revert, but would double check.

Reply:
we arent using 0.8 though, right?  

Why aren't we using sub(), add() etc from StrategyMath?

Reply:
Changed to `sub`